### DISCUSSION TITLE
TinyUSB debugging methods...

### DISCUSSION THREAD

**Discussion by achieve4s** [hathach/tinyusb#2348] [General]
Hello,

I followed the build steps described in the web page and the examples are successfully built and run on the target of i.mxrt 1064 evk. The environment is Ubuntu. I used the JLink to flash the image to the target without any issue.

I am wondering:
(1) How is the debugging done in the TinyUSB development, JLink or any other tools/methods in some IDE or command line in Linux? 
(2) Can anyone share how you debug your TinyUSB applications in Linux?

Really appreciate it!

**Comment by hathach** [comment]
I also clion as IDE, and it has great support for debugging. Though you can use any IDE, command line debugger that support make/cmake build.

**Reply by achieve4s** [reply]
Hi Hathach,

Thanks for your reply and sorry for the late response...having been sick for a couple of weeks :-( .

Is the procedure documented somewhere to import the TinyUSB to IDE like CLion for building and debugging? I did quite a bit searching but did not find anything related.

I am also wondering how to use VS Code with cmake plug-in for building/debugging TinyUSB...

**Reply by hathach** [reply]
To use with tinyusb example, load the examples/CMakeLists.txt then define your cmake profile with `-DBOARD=yourboard` . For how to use/debug with clion, check out jetbrain documentation
